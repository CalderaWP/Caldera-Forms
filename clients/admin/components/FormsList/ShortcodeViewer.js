import React from 'react';
import propTypes from 'prop-types'

/**
 * Show shortcode viewer UI
 * @param props
 * @return {*}
 * @constructor
 */
export const ShortcodeViewer = (props) => {
	if (true === props.show) {
		return (
			<span>

				<span>
					[caldera_forms="{props.formId}"]
				</span>
				<button
					className="button cf-form-shortcode-preview-button"
					onClick={props.onButtonClick}
				>
				Close
			</button>
			</span>

		)
	}
	return (
		<button
			className="button cf-form-shortcode-preview-button"
			onClick={props.onButtonClick}
		>
			Get Shortcode
		</button>
	)
}

ShortcodeViewer.propTypes = {
	formId: propTypes.string.isRequired,
	onButtonClick: propTypes.func.isRequired,
	show: propTypes.bool
};

ShortcodeViewer.defaultProps = {
	show: false
}



