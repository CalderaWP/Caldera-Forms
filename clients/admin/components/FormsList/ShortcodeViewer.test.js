import renderer from "react-test-renderer";
import React from 'react';
import {ShortcodeViewer} from "./ShortcodeViewer";


describe('ShortcodeViewer component', () => {
	it('Shows inital view', () => {
		const formList = renderer.create(
			<ShortcodeViewer
				formId={'cf1'}
				onButtonClick={() => {
				}}
			/>
		);
		expect(formList.toJSON()).toMatchSnapshot();
	});

	it('Shows shorcode actually', () => {
		const formList = renderer.create(
			<ShortcodeViewer
				formId={'cf1'}
				onButtonClick={() => {
				}}
				show={true}
			/>
		);
		expect(formList.toJSON()).toMatchSnapshot();
	});

});