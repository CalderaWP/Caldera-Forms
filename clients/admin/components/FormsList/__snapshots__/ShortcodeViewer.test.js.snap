// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortcodeViewer component Shows inital view 1`] = `
<button
  className="button cf-form-shortcode-preview-button"
  onClick={[Function]}
>
  Get Shortcode
</button>
`;

exports[`ShortcodeViewer component Shows shorcode actually 1`] = `
<span>
  <span>
    [caldera_forms="
    cf1
    "]
  </span>
  <button
    className="button cf-form-shortcode-preview-button"
    onClick={[Function]}
  >
    Close
  </button>
</span>
`;
